### ISSUE TITLE
🐛 Card repeater action item labels are 0 indexed

### ISSUE THREAD

**Issue by esizer** [GCTC-NTGC/gc-digital-talent#12354]
## 🐛 Bug

The action items for our card repeater (edit, remove) are 0 indexed so may cause issues for screen readers.

## 🦋 Expected Behaviour

References to items are 1 based indexed and not 0 based.

## 📋 Steps to Reproduce

1. Login as applicant `applicant@test.com`
2. Navigate to a skill showcase `/applicant/skills/showcase/top-5-behavioural-skills`
3. Add a skill to show the delete button
4. Open the accessibility tree inspector
5. Inspect the delete button
6. Observe the name being "Remove item 0" on the first item

## 📸 Screenshot

![2024-12-19_10-40](https://github.com/user-attachments/assets/01756f44-0986-4be4-95a5-31b51115b73e)

## 🙋‍♀️ Proposed Solution

Increment the index for the action buttons by 1.

## ✅ Acceptance Criteria

- [ ] Actions buttons are 1 based indexed

**Comment by mnigh** [comment]
This seems to be only a problem on the `SkillShowcaseCard`. The `AssessmentStepCard` and the `GeneralQuestionCard` that both use the underlying `CardRepeater.Card` component do not have this problem. Those components both use `CardRepeater.Remove` whereas `SkillShowcaseCard` uses `RemoveDialog`, 🤔.


## Screenshots
<img width="1334" alt="Image" src="https://github.com/user-attachments/assets/daf75e6f-386c-4252-9a68-94a9b094670d" />
<img width="1331" alt="Image" src="https://github.com/user-attachments/assets/b427a293-94ed-4e4a-adc6-aac51acab469" />

**Comment by esizer** [comment]
> Instead of `Remove item {index}` that provides no context for a screen reader, wouldn't `Remove {skillName} from your top 5 behavioural skills` be better? The context would be different for each one:

I like the idea of improving the button text and would like to see it extend to the heading or content of the dialog that opens as well.

**Comment by NienkeBr** [comment]
Hi, yes please, let's update the button text and the dialog as well. the dialog heading can be the same for all and I would change the content to specify which skill and from which subsection of your skill showcase:

> **Remove skill from your showcase**
> You’re about to remove {skillname} from your top 5 behavioural skills. 
> Are you sure you want to continue?
> [Remove] [Cancel and go back]

> **Remove skill from your showcase**
> You’re about to remove {skillname} from your top 10 technical skills. 
> Are you sure you want to continue?
> [Remove] [Cancel and go back]

> **Remove skill from your showcase**
> You’re about to remove {skillname} from the behavioural skills you’d like to improve. 
> Are you sure you want to continue?
> [Remove] [Cancel and go back]

> **Remove skill from your showcase**
> You’re about to remove {skillname} from the skills you’d like to learn. 
> Are you sure you want to continue?
> [Remove] [Cancel and go back]

Could we also move the remove button to the left and make the cancel and go back button a text-only link?

![Image](https://github.com/user-attachments/assets/94eb85cc-d03c-4219-96ac-f628f29919a5)

**Comment by esizer** [comment]
> Hi, yes please, let's update the button text and the dialog as well. the dialog heading can be the same for all and I would change the content to specify which skill and from which subsection of your skill showcase:
> 
> > **Remove skill from your showcase**
> > You’re about to remove {skillname} from your top 5 behavioural skills.
> > Are you sure you want to continue?
> > [Remove] [Cancel and go back]
> 
> > **Remove skill from your showcase**
> > You’re about to remove {skillname} from your top 10 technical skills.
> > Are you sure you want to continue?
> > [Remove] [Cancel and go back]
> 
> > **Remove skill from your showcase**
> > You’re about to remove {skillname} from the behavioural skills you’d like to improve.
> > Are you sure you want to continue?
> > [Remove] [Cancel and go back]
> 
> > **Remove skill from your showcase**
> > You’re about to remove {skillname} from the skills you’d like to learn.
> > Are you sure you want to continue?
> > [Remove] [Cancel and go back]
> 
> Could we also move the remove button to the left and make the cancel and go back button a text-only link?
> 
> ![Image](https://github.com/user-attachments/assets/94eb85cc-d03c-4219-96ac-f628f29919a5)

Along the same lines, I beleive this should be an [alert dialog](https://61e099a1bb1465003a73d3ce-kahmcamgke.chromatic.com/?path=/story/ui-src-components-alertdialog--default&globals=theme:GCDT+Dark) and not a normal dialog as we are [trying to elicit a response out of the user as to whether they would like to take an action or not](https://www.w3.org/WAI/ARIA/apg/patterns/alertdialog/#:~:text=An%20alert%20dialog%20is%20a%20modal%20dialog%20that%20interrupts%20the%20user%27s%20workflow%20to%20communicate%20an%20important%20message%20and%20acquire%20a%20response.).

**Comment by petertgiles** [comment]
@esizer it looks like the discussion is suggesting a fairly large scope change.  When you have some time could you please update the issue text to reflect what you'd like to see done?

**Comment by petertgiles** [comment]
Decision in refinement : we're going to keep the scope really small right now.  Just change the 0-based description to a 1-based description.
